### PR TITLE
Misc Fixes

### PR DESCRIPTION
**Pull request by Saito720**
Removed AMFP color grading for MinGW since it's in an unfinished state. Turned off USERDIR_RESOURCES and set the CXX standard to 14 for MinGW.

CXX 14 works without issue. It's the definition just prior to 17. CXX 17 introduced the byte definition conflicts.

### CODE REVIEW

**Comment by jkulawik** [comment]
What was the issue with color grading? We don't need it (in fact it would probably be better to review and test that whole feature) but I'm curious.

I'd suggest a few changes:

* Nitpick, but it would be clearer if USERDIR_RESOURCES was a single option and then was `set` if `MINGW` is on
* Instead of using `__MINGW32__` for the colorgrading stuff, it would be good to make that an option and then set it to false for MinGW. So something like this:

```cmake
option(COLORGRADING_SUPPORT "Add support for color grading areas during compilation" OFF)

if(MINGW)
    set(COLORGRADING_SUPPORT OFF) // or something like this
endif()

// in the HPL2 CMake
if(COLORGRADING_SUPPORT)
    add_compile_definitions(COLORGRADING_SUPPORT)
endif()
```
On top of that, it would probably be good to check if there is no colorgrading files being added when it's not used. See how with FBX some files are only added if the option is on.

and then in code (note that you'd have to change `ifndef` do `ifdef`):
```cpp
#ifdef COLORGRADING_SUPPORT
```

File: amnesia/src/game/LuxMapHandler.cpp [hunk 1]
```diff
@@ -395,7 +397,7 @@ void cLuxMapHandler::ChangeMap(const tString &asMapName, const tString &asStartP
 cLuxMap *cLuxMapHandler::LoadMap(const tString &asFileName, bool abLoadEntities) {
 	const tString & mapname = FileToMapName(asFileName);
 
-	gpBase->mpEffectHandler->GetColorGrading()->InitializeLUT(mapname + "_colorgrading.png");
+	//gpBase->mpEffectHandler->GetColorGrading()->InitializeLUT(mapname + "_colorgrading.png");
```
Comment (jkulawik) [inline amnesia/src/game/LuxMapHandler.cpp:400]: Why is this commented out instead of using a preprocessor check?

Comment (Saito720) [reply amnesia/src/game/LuxMapHandler.cpp:400]: Commented out for the aforementioned reasons.

Comment (jkulawik) [reply amnesia/src/game/LuxMapHandler.cpp:400]: The bottom line is that *theoretically* the color grading areas should work without this.

**Pull request merged**